adapter/PreciceInterface.c

```diff
@@ -420,6 +420,7 @@ void Precice_FreeData( SimulationData * sim )
 		if( sim->preciceInterfaces[i] != NULL ){
 			free( sim->preciceInterfaces[i] );
 		}
+		free( sim->preciceInterfaces );
```

Comment:
This should be outside of the loop :wink:

Reply:
oh, indeed, there is a `for` loop here! :see_no_evil:

Reply:
Wooops.
Surprised it didn't blow then. Is that undefined behavior when you free the same thing multiple times ?

Reply:
Hum now that I think about it, I tested locally then copy-pasted on Github instead of a  true commit from source, so it's probably just a typo. I'll run again to be sure.